### ISSUE TITLE
Choose-whens are not tested in sequential order

### ISSUE THREAD

**Issue by RacingTadpole** [AlexGilleran/jsx-control-statements#37]
Eg.
```
    <Choose>
        <When condition={this.props.when1}>
            <span>WhenBlock1</span>
        </When>
        <When condition={this.props.when2}>
            <span>WhenBlock2</span>
        </When>
        <Otherwise>
            <span>OtherwiseBlock</span>
        </Otherwise>
    </Choose>
```
    it('should render the first block when both conditions true', function () {
      var rendered = util.render(Fixture, {when1: true, when2: true});
      expect(rendered).to.match(util.matchTextWithinSpan('WhenBlock1'));
    });

fails - it renders `WhenBlock2`.

**Comment by AlexGilleran** [comment]
Thanks @RacingTadpole - is it always when2? I have a feeling it's actually going in _reverse_ sequential order.

**Comment by RacingTadpole** [comment]
Yes - it looks like it is, maybe due to this line: https://github.com/AlexGilleran/jsx-control-statements/blob/master/src/chooseStatement.js#L72 . When I remove the `reverse()` all the tests still pass, and the test above passes too. I'll submit a PR for that.

We could just define the reverse behaviour to be correct, but that's pretty different from what you'd expect from the analogy with "if-elseif-else". (Eg. the [original discussion](https://github.com/AlexGilleran/jsx-control-statements/issues/26#issuecomment-171342769) of this included an example a lot like this, with the "WhenBlock1" answer.)

**Comment by AlexGilleran** [comment]
Yeah agree re: the order.

Any thoughts @texttechne ?

**Comment by texttechne** [comment]
Ahh, I see my mistakes. The most important one is having missed to test ordering :smiley: 

The second mistake is all about reversed ordering. 

- The array of children is reduced from the right side (l.17), so the ordering of `when` blocks gets reversed. 
- This is the reason I intended to revert them in the mentioned line 72. 
- Actually this is wrong, because the ternary expression itself is built up from right to left (l.73), which requires reversed ordering...

So @RacingTadpole is completely right with his analysis and bugfix. :+1:

**Comment by texttechne** [comment]
We should leave this issue open until it is released.

**Comment by AlexGilleran** [comment]
Released as v3.1.1 :)

**Comment by texttechne** [comment]
:+1:
